From: win7-support-bot <win7@cromite.org>
Date: Sun, 26 Jul 2026 00:00:00 +0000
Subject: WIN7: chrome-browser / registry

Files:
 chrome/browser/shell_integration_win.cc

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
diff --git a/chrome/browser/shell_integration_win.cc b/chrome/browser/shell_integration_win.cc
index a5899b23e7e54..e2ed942a73937 100644
--- a/chrome/browser/shell_integration_win.cc
+++ b/chrome/browser/shell_integration_win.cc
@@ -37,7 +37,6 @@
 #include "base/time/time.h"
 #include "base/timer/timer.h"
 #include "base/win/registry.h"
-#include "base/win/scoped_co_mem.h"
 #include "base/win/scoped_propvariant.h"
 #include "base/win/shlwapi.h"
 #include "base/win/shortcut.h"
@@ -631,19 +630,12 @@ void MigrateChromeAndChromeProxyShortcuts(
 }
 
 std::wstring GetHttpSchemeUserChoiceProgId() {
-  Microsoft::WRL::ComPtr<IApplicationAssociationRegistration> registration;
-  HRESULT hr = ::SHCreateAssociationRegistration(IID_PPV_ARGS(&registration));
-  if (FAILED(hr)) {
-    return std::wstring();
-  }
-
-  base::win::ScopedCoMem<wchar_t> prog_id;
-  hr = registration->QueryCurrentDefault(L"http", AT_URLPROTOCOL, AL_EFFECTIVE,
-                                         &prog_id);
-  if (FAILED(hr)) {
-    return std::wstring();
-  }
-  return prog_id.get();
+  std::wstring prog_id;
+  base::win::RegKey key(HKEY_CURRENT_USER, ShellUtil::kRegVistaUrlPrefs,
+                        KEY_QUERY_VALUE);
+  if (key.Valid())
+    key.ReadValue(L"ProgId", &prog_id);
+  return prog_id;
 }
 
 }  // namespace

